woocommerce/woocommerce
# Incorporate blocking issues feedback on initial e2e environment package

### All Submissions:


### Changes proposed in this Pull Request:

- Software versions can now be specified with 
```
TRAVIS_MARIADB_VERSION=10.5.3 TRAVIS_PHP_VERSION=7.4.5 WP_VERSION=5.4.1 npm run docker:up
```
- Docker will download and cache a copy of libraries when requesting versions that haven't been used in Docker on a system before. (eg. When WordPress 5.5 is released a new Docker WordPress image will be downloaded.) You will see the downloads logged in the output.
- Ensure Docker is down when making changes to `tests/e2e/config/default.json`. 

### How to test the changes in this Pull Request:

1. Run `npm run build:packages`
2. Try `up` & `down` using combinations of environment variables above including with none set
3. `npm run test:e2e`
4. Copy `tests/e2e/env/config/default.json` to `tests/e2e/config/default.json`
5. Edit `tests/e2e/config/default.json` and change the port in the `url` value
6. `npm run docker:up` and `npm run test:e2e`.
7. Create the following `/tests/e2e/config/jest-puppeteer.config.js`
```
const { useE2EJestPuppeteerConfig } = require( '@woocommerce/e2e-environment' );

const puppeteerConfig = useE2EJestPuppeteerConfig( {
	launch: {
		headless: false,
	}
} );

module.exports = puppeteerConfig;
```
8. `npm run test:e2e`
9. Tests should run with browser (some may fail for timing)
10. Change `headless` to `true`
11. `npm run test:e2e-dev` should run without browser
12. Replace `/tests/e2e/config/jest.config.js` in this branch with the version in `master`
13. `npm run test:e2e`

### Other information:




### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.
